pkg/scaffold/template.go

```diff
@@ -76,20 +80,89 @@ func (t Template) Render(f function.Function, step function.Step) error {
 		return fmt.Errorf("error creating function directory: %w", err)
 	}
 
+	if err := t.RenderTriggers(ctx, f); err != nil {
+		return err
+	}
+	for _, s := range f.Steps {
+		if err := t.RenderStep(ctx, f, s); err != nil {
+			return err
+		}
+	}
+
+	// Once complete, state should contain everything we need to create our
+	// function file.
+	byt, err := function.MarshalJSON(f)
+	if err != nil {
+		return fmt.Errorf("error creating JSON: %s", err)
+	}
+
+	if err := os.WriteFile(filepath.Join(root, "inngest.json"), byt, 0644); err != nil {
+		return fmt.Errorf("Error writing inngest.json: %s", err)
+	}
+
+	return nil
+}
+
+func (t Template) RenderTriggers(ctx context.Context, f function.Function) error {
+	root := t.Root(f)
+
+	if err := upsertDir(filepath.Join(root, "events")); err != nil {
+		return fmt.Errorf("error making events directory: %w", err)
+	}
+
+	// For each event within the function create a new event file.
+	for n, trigger := range f.Triggers {
+		if trigger.EventTrigger == nil {
+			continue
+		}
+
+		if trigger.EventTrigger.Definition == nil || trigger.EventTrigger.Definition.Def == "" {
+			// Use an empty event format.
+			trigger.EventTrigger.Definition = &function.EventDefinition{
+				Format: function.FormatCue,
+				Synced: false,
+				Def:    fmt.Sprintf(evtDefinition, strconv.Quote(trigger.Event)),
+			}
+		}
+
+		cue, err := trigger.Definition.Cue(ctx)
+		if err != nil {
+			// XXX: We would like to log this as a warning.
+			continue
+		}
+
+		name := fmt.Sprintf("%s.cue", eventFilename(trigger.Event))
+		path := filepath.Join(root, "events", name)
```

Comment:
Can we move this `"events"` string to be a `const` or `var` somewhere? It's a nit, but I'm thinking that the higher it is, the more it becomes a standard.